config/params/config_utils_develop.go

```diff
@@ -9,33 +9,32 @@ import (
 	"github.com/mohae/deepcopy"
 )
 
-var beaconConfig = MainnetConfig()
-var beaconConfigLock sync.RWMutex
+var cfgrw sync.RWMutex
 
 // BeaconConfig retrieves beacon chain config.
 func BeaconConfig() *BeaconChainConfig {
-	beaconConfigLock.RLock()
-	defer beaconConfigLock.RUnlock()
-	return beaconConfig
+	cfgrw.RLock()
+	defer cfgrw.RUnlock()
+	return Registry.GetActive()
 }
 
 // OverrideBeaconConfig by replacing the config. The preferred pattern is to
 // call BeaconConfig(), change the specific parameters, and then call
 // OverrideBeaconConfig(c). Any subsequent calls to params.BeaconConfig() will
 // return this new configuration.
 func OverrideBeaconConfig(c *BeaconChainConfig) {
-	beaconConfigLock.Lock()
-	defer beaconConfigLock.Unlock()
-	beaconConfig = c
+	cfgrw.Lock()
+	defer cfgrw.Unlock()
+	Registry.active = c
 }
 
 // Copy returns a copy of the config object.
 func (b *BeaconChainConfig) Copy() *BeaconChainConfig {
-	beaconConfigLock.RLock()
-	defer beaconConfigLock.RUnlock()
+	cfgrw.RLock()
+	defer cfgrw.RUnlock()
 	config, ok := deepcopy.Copy(*b).(BeaconChainConfig)
 	if !ok {
-		config = *beaconConfig
+		panic("somehow deepcopy produced a BeaconChainConfig that is not of the same type as the original")
```

Comment:
previously in the case where this somehow fails (I suspect it *never* fails, because that would require deepcopy to suddenly return a different type than it as given) the return of the function would be whatever beaconConfig happens to currently be, rather than what was passed in! I think it would be better if this just fails spectacularly, rather than giving a different config.